### DISCUSSION TITLE
How to exclude modules and modes while compiling with PlatformIO ?

### DISCUSSION THREAD

**Discussion by tk5ep** [jgromes/RadioLib#933] [Q&A]
Hi,
To speed up the compiling, I tried to exclude several options I don't need with help of the platformIO.ini.
I have added : 

build_flags =
-DRADIOLIB_EXCLUDE_CC1101
-DRADIOLIB_EXCLUDE_NRF24
etc...

But that doesn't seem to work.
Did I overlook something ? I'm not a great PlatformIO specialist. 

Thanks,

**Comment by jgromes** [comment]
The macro checks test for value, not the macro presence, so you should use `-DRADIOLIB_EXCLUDE_CC1101=1` etc.

However, it's probably not worth it for you. I don't know what sort of MCU you have, but if it's something like ESP32, RadioLib sources will be <10% of the code base that needs to be compiled, so the improvement will be very limited. The primary use case for these macros is to disable compilation in cases where it doesn't make sense, e.g. attempting to build the STM32WL module on microcontrollers that are not STM32WL.

**Comment by tk5ep** [comment]
Hi and thanks,
I will try asap. Now on the ferry boat for a few weeks on the mainland and
probably away from the keyboard.
Thanks again for this great library and support.
73 Patrick

Le jeu. 18 janv. 2024, 18:32, Jan Gromeš ***@***.***> a
écrit :
